src/DotVVM.Framework/Compilation/Static/StaticViewCompiler.cs

```diff
@@ -0,0 +1,277 @@
+        public static DotvvmConfiguration CreateConfiguration(
+            Assembly dotvvmProjectAssembly,
+            string dotvvmProjectDir)
+        {
+            InitializeDotvvmControls(dotvvmProjectAssembly);
+            return ConfigurationInitializer.GetConfiguration(dotvvmProjectAssembly, dotvvmProjectDir, services =>
+            {
+                services.AddSingleton<IControlResolver, StaticViewControlResolver>();
+                services.TryAddSingleton<IViewModelProtector, FakeViewModelProtector>();
+                services.AddSingleton(new RefObjectSerializer());
+                // Yes, this is here so that there can be a circular dependency in StaticViewControlResolver.
+                // I'm not happy about it, no, but the alternative is a more-or-less complete rewrite.
+                services.AddSingleton(p => new StaticViewCompiler(
+                    p.GetRequiredService<DotvvmConfiguration>(),
+                    dotvvmProjectAssembly));
+
+                // HACK: IDotvvmCacheAdapter is not in v2.0.0 that's why it's hacked this way.
+                var iCacheAdapter = Type.GetType(IDotvvmCacheAdapterName);
+                if (iCacheAdapter is object)
+                {
+                    services.AddSingleton(iCacheAdapter, Type.GetType(SimpleDictionaryCacheAdapterName));
+                }
+
+                // TODO: Uncomment when the views can actually be compiled into one assembly.
+                // var bindingCompiler = new AssemblyBindingCompiler(
+                //     assemblyName: null,
+                //     className: null,
+                //     outputFileName: null,
+                //     configuration: null);
+                // services.AddSingleton<IBindingCompiler>(bindingCompiler);
+                // services.AddSingleton<IExpressionToDelegateCompiler>(bindingCompiler.GetExpressionToDelegateCompiler());
+            });
+        }
+
+        public StaticView GetView(string viewPath)
+        {
+            if (viewCache.ContainsKey(viewPath))
+            {
+                return viewCache[viewPath];
+            }
+
+            var view = CompileView(viewPath);
+            // NB: in the meantime, the view could have been compiled on another thread
+            return viewCache.GetOrAdd(viewPath, view);
+        }
+
+        public IEnumerable<StaticView> CompileAllViews()
+        {
+            var markupControls = CompileViews(configuration.Markup.Controls.Select(c => c.Src));
+            var views = CompileViews(configuration.RouteTable.Select(r => r.VirtualPath));
+            return markupControls.Concat(views).ToImmutableArray();
+        }
+
+        private ImmutableArray<StaticView> CompileViews(IEnumerable<string?> paths)
+        {
+            return paths
+                .Where(p => !string.IsNullOrWhiteSpace(p))
+                .Select(p => GetView(p!))
+                .ToImmutableArray();
+        }
+
+        private StaticView CompileView(string viewPath)
```

Comment:
Oh, this looks like a partially broken duplication of `DefaultViewCompiler` - could we merge the logic or is it unused code anyway? In particular, this probably breaks on recursive markup controls, and it does not run all visitors.

Reply:
I agree that the duplication was really bad, so I removed it. However, not the linter is limited to one parser error per file. We should look into that at some point.